mm2src/coins/utxo/utxo_tests.rs

```diff
@@ -4265,3 +4265,90 @@ fn test_utxo_validate_valid_and_invalid_pubkey() {
     assert!(coin.validate_other_pubkey(&[1u8; 20]).is_err());
     assert!(coin.validate_other_pubkey(&[1u8; 8]).is_err());
 }
+
+#[test]
+fn test_block_header_utxo_loop() {
+    use crate::utxo::utxo_builder::{block_header_utxo_loop, BlockHeaderUtxoLoopExtraArgs};
+    use futures::future::{Either, FutureExt};
+
+    static mut CURRENT_BLOCK_COUNT: u64 = 150;
+
+    ElectrumClient::get_block_count
+        .mock_safe(move |_| MockResult::Return(Box::new(futures01::future::ok(unsafe { CURRENT_BLOCK_COUNT }))));
+
+    BlockHeaderUtxoLoopExtraArgs::default.mock_safe(move || {
+        MockResult::Return(BlockHeaderUtxoLoopExtraArgs {
+            chunk_size: 100,
+            error_sleep: 5.,
+            success_sleep: 1.,
+        })
+    });
+
+    let block_header_storage = BlockHeaderStorageForTests::new(TEST_COIN_NAME.to_string());
+    BlockHeaderStorage::new_from_ctx.mock_safe(move |_, _| {
+        MockResult::Return(Ok(BlockHeaderStorage {
+            inner: Box::new(block_header_storage.clone()),
+        }))
+    });
```

Comment:
I just realized that we can avoid adding a new `BlockHeaderStorageForTests`.
My initial idea was to create a shared `block_header_storage` variable, put some block headers there and then run the test from a specified block. But you followed another way (I like it much more than my way):
You create an in-memory storage and start the test with the empty storage.

As an alternative to the `BlockHeaderStorageForTests` we can use a general `SqliteBlockHeadersStorage`. As you can see below, the storage is initialized with a context's SQLite connection.
https://github.com/KomodoPlatform/atomicDEX-API/blob/9e9b13a2e6a0d762897941e1882b23f501ff0072/mm2src/coins/utxo/utxo_block_header_storage/mod.rs#L31-L36
To make it in-memory, we can use `mm_ctx_with_custom_db`:
https://github.com/KomodoPlatform/atomicDEX-API/blob/9e9b13a2e6a0d762897941e1882b23f501ff0072/mm2src/mm2_test_helpers/src/for_tests.rs#L425-L434

Summing up, we can do the following:
1. Remove `BlockHeaderStorageForTests`
2. Undo `BlockHeaderStorage::new_from_ctx` mockable
3. Create an MmCtx with `mm_ctx_with_custom_db` in the test
4. Pass the `mm_ctx` to `UtxoArcBuilder::new`

Reply:
clear..thanks